src/gt4py/next/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -517,15 +490,58 @@ def remap_args(s: Scan) -> Scan:
                 res.append(execution)
         return res
 
-    def visit_FencilDefinition(
-        self, node: itir.FencilDefinition, **kwargs: Any
-    ) -> FencilDefinition:
+    def visit_Stmt(self, node: itir.Stmt, **kwargs: Any) -> None:
+        raise AssertionError("Internal error: all Stmts need to be handled explicitly.")
+
+    def visit_SetAt(
+        self, node: itir.SetAt, *, extracted_functions: list, **kwargs: Any
+    ) -> Union[StencilExecution, ScanExecution]:
+        assert _is_applied_apply_stencil(node.expr)
+        stencil = node.expr.fun.args[0]  # type: ignore[attr-defined] # checked in assert
+        domain = node.domain
+        inputs = node.expr.args
+        backend = Backend(domain=self.visit(domain, stencil=stencil, **kwargs))
+        if _is_scan(stencil):
+            scan_id = self.uids.sequential_id(prefix="_scan")
+            scan_lambda = self.visit(stencil.args[0], **kwargs)
+            forward = _bool_from_literal(stencil.args[1])
+            scan_def = ScanPassDefinition(
+                id=scan_id, params=scan_lambda.params, expr=scan_lambda.expr, forward=forward
+            )
+            extracted_functions.append(scan_def)
+            scan = Scan(
+                function=SymRef(id=scan_id),
+                output=0,
+                inputs=[i + 1 for i, _ in enumerate(inputs)],
+                init=self.visit(stencil.args[2], **kwargs),
+            )
+            column_axis = self.column_axis
+            assert isinstance(column_axis, common.Dimension)
+            return ScanExecution(
+                backend=backend,
+                scans=[scan],
+                args=[self._visit_output_argument(node.target), *self.visit(inputs)],
+                axis=SymRef(id=column_axis.value),
+            )
+        return StencilExecution(
+            stencil=self.visit(
+                stencil,
+                force_function_extraction=True,
+                extracted_functions=extracted_functions,
+                **kwargs,
+            ),
+            output=self._visit_output_argument(node.target),
+            inputs=self.visit(inputs, **kwargs),
+            backend=backend,
+        )
+
+    def visit_Program(self, node: itir.Program, **kwargs: Any) -> FencilDefinition:
         extracted_functions: list[Union[FunctionDefinition, ScanPassDefinition]] = []
-        executions = self.visit(node.closures, extracted_functions=extracted_functions)
+        executions = self.visit(node.body, extracted_functions=extracted_functions)
         executions = self._merge_scans(executions)
         function_definitions = self.visit(node.function_definitions) + extracted_functions
         offset_definitions = {
-            **_collect_dimensions_from_domain(node.closures),
+            **_collect_dimensions_from_domain(node.body),
             **_collect_offset_definitions(node, self.grid_type, self.offset_provider),
         }
         return FencilDefinition(
```

Comment:
I think it makes sense to rename `gtfn_ir.FencilDefinition` to `gtfn_ir.Program`.

Reply:
done